velox/exec/RowContainer.cpp

```diff
@@ -408,6 +422,38 @@ void RowContainer::freeAggregates(folly::Range<char**> rows) {
   }
 }
 
+void RowContainer::freeNextRowVectors(folly::Range<char**> rows, bool clear) {
+  if (!nextOffset_ || !hasDuplicateRows_) {
+    return;
+  }
+
+  for (auto row : rows) {
+    auto& vector = getNextRowVector(row);
+    if (vector) {
+      if (clear) {
+        // Clear all rows, we can clear the nextOffset_ slots and delete the
+        // next-row-vector.
+        for (auto& next : *vector) {
+          getNextRowVector(next) = nullptr;
+        }
+        delete vector;
+      } else {
+        auto iter = std::find(vector->begin(), vector->end(), row);
+        VELOX_CHECK(iter != vector->end());
+        vector->erase(iter);
+        vector = nullptr;
```

Comment:
If `row` is the last element in `vector`, we should also release `vector` otherwise it's memory leak

Reply:
Thanks, good catch! Updated test case to cover this.

Reply:
I checked internally and do not find directly reference to `eraseRows`.  Maybe we can make the assumption (and add check) that when this is called, all the rows belong to the same key will be erased.  This way should make it much more efficient.